app/code/Magento/CatalogRule/Pricing/Price/CatalogRulePrice.php

```diff
@@ -99,9 +99,9 @@ public function getValue()
                         $this->product->getId()
                     );
                 $this->value = $this->value ? floatval($this->value) : false;
-                if ($this->value) {
-                    $this->value = $this->priceCurrency->convertAndRound($this->value);
-                }
+            }
+            if ($this->value) {
```

Comment:
```
if ($this->product->hasData(self::PRICE_CODE)) {
    $this->value = floatval($this->product->getData(self::PRICE_CODE)) ?: false;
```

Where this value is stored? I believe currency conversion is missed there.

So, it's better to store totally prepared value than convert it after.

However, if the value stored in product is not store-aware by design, fix looks ok to me. To avoid regression it's better to create proper integration test, with catalog rule and multiple currencies, as change of started to fail unit test does not enforce anything (i.e. somebody doing next changes will just do the same - change hardcoded expectations - as test case is not sticked to any concrete scenario).